test/integration/targets/inventory_aws_conformance/runme.sh

```diff
@@ -0,0 +1,152 @@
+  ec2_eventsSet: backwardscompat | default("")  # FIXME would need to add a describe_instance_status API call for instances to be able to add an events hostvar
+  ec2_persistent: backwardscompat | default(false)  # FIXME would need to add a describe_spot_instance_requests API call for any instances that have the SpotInstanceRequestId variable
+
+  # vars that change
+  ec2_block_devices: dict(block_device_mappings | map(attribute='device_name') | list | zip(block_device_mappings | map(attribute='ebs.volume_id') | list))
+  ec2_dns_name: public_dns_name
+  ec2_group_name: placement.group_name
+  ec2_id: instance_id
+  ec2_instance_profile: iam_instance_profile | default("")
+  ec2_ip_address: public_ip_address
+  ec2_kernel: kernel_id | default("")
+  ec2_monitored:  monitoring.state in ['enabled', 'pending']
+  ec2_monitoring_state: monitoring.state
+  ec2_owner_id: account_id
+  ec2_placement: placement.availability_zone
+  ec2_ramdisk: ramdisk_id | default("")
+  ec2_reason: state_transition_reason
+  ec2_security_group_ids: security_groups | map(attribute='group_id') | list |  join(',')
+  ec2_security_group_names: security_groups | map(attribute='name') | list |  join(',')
+  ec2_state: state.name
+  ec2_state_code: state.code
+  ec2_state_reason: state_reason.message if state_reason is defined else ""
+  ec2_sourceDestCheck: source_dest_check | lower | string  # butchered snake_case case not a typo.
+
+  # vars that just need ec2_ prefix
+  ec2_ami_launch_index: ami_launch_index | string
+  ec2_architecture: architecture
+  ec2_client_token: client_token
+  ec2_ebs_optimized: ebs_optimized
+  ec2_hypervisor: hypervisor
+  ec2_image_id: image_id
+  ec2_instance_type: instance_type
+  ec2_key_name: key_name
+  ec2_launch_time: 'launch_time | regex_replace(" ", "T") | regex_replace("(\+)(\d\d):(\d)(\d)$", ".\g<2>\g<3>Z")'
+  ec2_platform: platform | default("")
+  ec2_private_dns_name: private_dns_name
+  ec2_private_ip_address: private_ip_address
+  ec2_public_dns_name: public_dns_name
+  ec2_region: placement.region
+  ec2_root_device_name: root_device_name
+  ec2_root_device_type: root_device_type
+  ec2_spot_instance_request_id: spot_instance_request_id
+  ec2_subnet_id: subnet_id
+  ec2_virtualization_type: virtualization_type
+  ec2_vpc_id: vpc_id
+  tags: dict(tags.keys() | map('regex_replace', '[^A-Za-z0-9\_]', '_') | list | zip(tags.values() | list))
+
+keyed_groups:
+  - key: '"ec2"'
+    separator: ""
+  - key: 'instance_id'
+    separator: ""
+  - key: tags
+    prefix: tag
+  - key: key_name
+    prefix: key
+  - key: placement.region
+    separator: ""
+  - key: placement.availability_zone
+    separator: ""
+  - key: platform
+    prefix: platform
+  - key: vpc_id
+    prefix: vpc_id
+  - key: instance_type
+    prefix: type
+  - key: image_id
+    separator: ""
+  - key: 'security_groups | json_query("[].name")'
+    prefix: security_group
```

Comment:
I don't know how much you talked to Elijah about this, so I could be repeating things, but we were throwing around the idea that we would test the _current_ values that the plugin returns.

I don't necessarily think there's value in testing this complex `compose` and `keyed_group` syntax, because if the source variables are there, there shouldn't be a problem.

We have a list of "favorite variables" that we are going to check for in integration testing. Those are in the old names now, but that set of variables can be trivially converted to a set of new variables.

The list of new variables we would want assertions on would include vars that keyed_groups sourced. So `security_groups[0]['name']`, for example, would be one. Meaning, we could assert that hostvars from the plugin has the security_groups list with at least 1 entry, which has a name.

Ideally, those supported variables would go into Ansible core documentation, they would be copied over, and would be surfaced in the UI of AWX / Tower... eventually.

Reply:
We have no guarantee that -any- source variables won't disappear or change with every future PR. This test asserts that doesn't happen.